src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -457,80 +421,43 @@ private IEnumerable<(ManifestId id, ManifestVersion version, SdkFeatureBand feat
 
         private string GetAdvertisingWorkloadsFilePath(SdkFeatureBand featureBand) => GetAdvertisingWorkloadsFilePath(_userProfileDir, featureBand);
 
         private static string GetAdvertisingWorkloadsFilePath(string userProfileDir, SdkFeatureBand featureBand) => Path.Combine(userProfileDir, $".workloadAdvertisingUpdates{featureBand}");
 
-        private async Task<string> GetOnlinePackagePath(SdkFeatureBand sdkFeatureBand, ManifestId manifestId, bool includePreviews)
-        {
-            string packagePath = await _nugetPackageDownloader.DownloadPackageAsync(
-                _workloadManifestInstaller.GetManifestPackageId(manifestId, sdkFeatureBand),
-                packageSourceLocation: _packageSourceLocation,
-                includePreview: includePreviews);
-
-            return packagePath;
-        }
-
-        private string GetOfflinePackagePath(SdkFeatureBand sdkFeatureBand, ManifestId manifestId, DirectoryPath? offlineCache = null)
-        {
-            string packagePath = Directory.GetFiles(offlineCache.Value.Value)
-                .Where(path => path.EndsWith(".nupkg"))
-                .Where(path =>
-                {
-                    var manifestPackageId = _workloadManifestInstaller.GetManifestPackageId(manifestId, sdkFeatureBand).ToString();
-                    return Path.GetFileName(path).StartsWith(manifestPackageId, StringComparison.OrdinalIgnoreCase);
-                })
-                .Max();
-
-            return packagePath;
-        }
-
-        private async Task<(bool, string)> GetManifestPackageUpdate(SdkFeatureBand sdkFeatureBand, ManifestId manifestId, bool includePreviews, DirectoryPath? offlineCache = null)
+        private async Task<(bool Success, string PackagePath)> GetManifestPackageUpdate(SdkFeatureBand sdkFeatureBand, ManifestId manifestId, bool includePreviews, DirectoryPath? offlineCache = null)
         {
-            if (offlineCache == null || !offlineCache.HasValue)
+            var manifestPackageId = _workloadManifestInstaller.GetManifestPackageId(manifestId, sdkFeatureBand);
+            if (offlineCache != null)
             {
-                try
-                {
-                    string packagePath = await GetOnlinePackagePath(sdkFeatureBand, manifestId, includePreviews);
-                    return (true, packagePath);
-                }
-                catch (NuGetPackageNotFoundException)
-                {
-                    return (false, null);
-                }
+                string offlinePackagePath = Directory.GetFiles(offlineCache.Value.Value)
+                    .Where(path => path.EndsWith(".nupkg") && Path.GetFileName(path).StartsWith(manifestPackageId.ToString(), StringComparison.OrdinalIgnoreCase))
+                    .Max();
+                return (Success: offlinePackagePath != null, PackagePath: offlinePackagePath);
```

Comment:
I'm wondering if `offlinePackagePath != null` should be `!string.IsNullOrEmpty(offlinePackagePath)`

That is a slight change in behavior, but it feels a little safer to me.

Reply:
I actually tested this before changing it because I questioned how `Max` works.
https://dotnetfiddle.net/ZjvAt1

The `Where` can only return non-empty strings of elements or `null`. It isn't possible for it to return empty string based on the condition set in the `Where`.